File: ruby/p001.rb
# Ruby Solution to Project Euler problem 1
# by Mychal Johnson
# https://github.com/mychalj/project-euler

# This creates an array with the numbers 1-999.
num = (1...1000).to_a

# The delete_if method deletes the numbers in the array that are not
# multiples of 3 or 5.
num.delete_if { |i| (i % 3).nonzero? && (i % 5).nonzero? }

# The inject method adds the individual values together in the array
# and stores the sum in the total variable.
total = num.inject(0) { |a, e| a + e }

# This outputs the total variable.
puts total

File: javascript/p001.js
/*
 * Javascript Solution to Project Euler problem 1
 * by Mychal Johnson
 *
 * https://github.com/mychalj/project-euler
 */

var num = [];

for(var i = 1; i < 1000; i++) {
  if (i % 3 === 0 || i % 5 === 0) {
    num.push(i);
  }
}

var total = num.reduce(function(total, num){return total + num;});

console.log(total);
